### ISSUE TITLE
Strix-Halo detection - "hw-detect"

### ISSUE THREAD

**Issue by FirebirdRender** [Pavelevich/llm-checker#75]
great tool, doesn't detect the VRAM correctly on AMD Strix-Halo correctly yet:

##DEBIAN 13
##AMD Strix-Halo

```
llm-checker hw-detect
llm-checker | Hardware Detection
------------------------------------------
⠋ Detecting hardware...WARNING: AMD GPU device(s) is/are in a low-power state. Check power control/runtime_status
WARNING: AMD GPU device(s) is/are in a low-power state. Check power control/runtime_status
WARNING: AMD GPU device(s) is/are in a low-power state. Check power control/runtime_status
WARNING: AMD GPU device(s) is/are in a low-power state. Check power control/runtime_status

✔ Hardware detected!

=== Hardware Detection ===
Summary:
  GFX Version: gfx1151 (2GB VRAM) + AMD RYZEN AI MAX+ 395 w/ Radeon 8060S
  Tier: ULTRA LOW
  Max model size: 0GB
  Best backend: AMD ROCm
  Dedicated GPUs: GFX Version: gfx1151
  Integrated GPUs: None

CPU:
  AMD RYZEN AI MAX+ 395 w/ Radeon 8060S
  Cores: 32 (16 physical)
  SIMD: AVX512
  [OK] AVX-512
  [OK] AVX2

ROCM:
  ROCm: 4.0.0
  Total VRAM: 2GB
  GFX Version:          gfx1151: 2GB

Fingerprint: rocm-gfx-version:-gfx1151-2gb
```

**I built a script for the host to show me on boot what the VRAM/GTT pool is, this might help:**

if [ -d /sys/module/amdgpu ]; then
    VENDOR="AMD"
    GTT_TOTAL_BYTES=$(cat /sys/module/amdgpu/drivers/pci:amdgpu/0000:*/mem_info_gtt_total 2>/dev/null || echo 0)

**Comment by FirebirdRender** [comment]
Proposed patch, adding support:

SYSTEM INFORMATION 
╭──────────────────────────────────────────────────
│ CPU: RYZEN AI MAX+ 395 w/ Radeon 8060S (32 cores, 3GHz)
│ Architecture: x64
│ RAM: 123GB
│ GPU: GFX Version: gfx1151
│ Backend: AMD ROCm
│ VRAM: N/AGB (Integrated)
│ Dedicated GPUs: None
│ Integrated GPUs: GFX Version: gfx1151
│ Hardware Tier: HIGH


[strix-halo-vram-fix.patch](https://github.com/user-attachments/files/26449139/strix-halo-vram-fix.patch)

**Comment by Pavelevich** [comment]
hi, is fixed, can u check if works, thx

**Comment by jedidja** [comment]
@Pavelevich @FirebirdRender I just pulled the source code and ran it on a Framework Desktop with 128 GB RAM (unified). This is the output:

```
~/source/Pavelevich/llm-checker ❯ node bin/enhanced_cli.js hw-detect

llm-checker | Hardware Detection
------------------------------------------
✔ Hardware detected!

=== Hardware Detection ===

Summary:
  GFX Version: gfx1151 (16GB VRAM) + AMD RYZEN AI MAX+ 395 w/ Radeon 8060S
  Tier: MEDIUM HIGH
  Max model size: 14GB
  Best backend: AMD ROCm
  Dedicated GPUs: GFX Version: gfx1151
  Integrated GPUs: None

CPU:
  AMD RYZEN AI MAX+ 395 w/ Radeon 8060S
  Cores: 32 (16 physical)
  SIMD: AVX512
  [OK] AVX-512
  [OK] AVX2

ROCM:
  ROCm: 3.1.0
  Total VRAM: 16GB
  GFX Version:          gfx1151: 16GB

Fingerprint: rocm-gfx-version:-gfx1151-16gb
```

Note that things _should_ be setup correctly:

```
~/source/Pavelevich/llm-checker ❯ amd-ttm
💻 Current TTM pages limit: 29360128 pages (112.00 GB)
💻 Total system memory: 125.07 GB
```

**Comment by jedidja** [comment]
Note that running the latest release, instead of building from code, shows this:

```
 SYSTEM INFORMATION
╭──────────────────────────────────────────────────
│ CPU: RYZEN AI MAX+ 395 w/ Radeon 8060S (32 cores, 3GHz)
│ Architecture: x64
│ RAM: 125GB
│ GPU: GFX Version: gfx1151
│ Backend: AMD ROCm
│ VRAM: 8GB (Dedicated)
│ Dedicated GPUs: GFX Version: gfx1151
│ Integrated GPUs: None
│ Hardware Tier: MEDIUM LOW
╰
```

So the patch has improved things a little bit, showing 16 GB VRAM instead of 8 GB but nowhere near what should be usable :)

**Comment by Pavelevich** [comment]
Thanks for the detailed retest, this is very helpful.

I checked the current source and there are effectively two different states right now:

1) In current source (`main`), Strix-Halo `gfx1151` is handled with a low-aperture safeguard, so it can report ~16GB instead of 2/8GB.
2) In the npm release path, you may still hit older behavior (8GB), which explains the mismatch you observed between source run and published package behavior.

So your report is valid: this is not fully solved yet for Strix-Halo unified memory.

The remaining issue is that this device is still being interpreted too much like dedicated VRAM instead of unified/shared memory for model-fit decisions. I’m keeping this issue open until we ship both:
- consistent release behavior (same as current source fixes), and
- proper integrated/unified handling for `gfx1151` path (including shared pool logic in effective memory/tier).

If you can, please also share these values from the same run:
- `rocm-smi --showmeminfo vram`
- `cat /sys/module/amdgpu/drivers/pci:amdgpu/0000:*/mem_info_gtt_total`
- `cat /sys/class/drm/card*/device/mem_info_vram_total`

Thanks again for the clear validation and for sticking with this.

**Comment by jedidja** [comment]
Sorry for the delay! Didn't check my notifications for a while.

```
============================ ROCm System Management Interface ============================
================================== Memory Usage (Bytes) ==================================
GPU[0]          : VRAM Total Memory (B): 536870912
GPU[0]          : VRAM Total Used Memory (B): 459845632
==========================================================================================
================================== End of ROCm SMI Log ===================================
```

```
120259084288
```

```
536870912
```

**Comment by signerless** [comment]
Fixed and merged in `main` via PR #84: https://github.com/Pavelevich/llm-checker/pull/84

What was done:
- Restored the canonical startup ASCII banner (the approved one) and bundled it in the package at `src/ui/banner-profesional-v2.txt`.
- Updated `src/ui/cli-theme.js` so the default banner source is the bundled file (no Desktop/iCloud fallback dependency).
- Removed the mascot/cat fallback path so that banner cannot reappear.
- Bumped package version to `3.5.14` and validated with test suite pass (34/34).

Thank you for opening the issue and helping catch this.

**Comment by signerless** [comment]
Fixed and merged in `main` via PR #84: https://github.com/Pavelevich/llm-checker/pull/84

What was done:
- Restored the canonical startup ASCII banner (the approved one) and bundled it in the package at `src/ui/banner-profesional-v2.txt`.
- Updated `src/ui/cli-theme.js` so the default banner source is the bundled file (no Desktop/iCloud fallback dependency).
- Removed the mascot/cat fallback path so that banner cannot reappear.
- Bumped package version to `3.5.14` and validated with test suite pass (34/34).

Thanks for opening the issue and helping catch this.
